r/R/feather.R

```diff
@@ -121,7 +121,8 @@ read_feather <- function(file, col_select = NULL, as_data_frame = TRUE, ...) {
   if (isTRUE(as_data_frame)) {
     out <- as.data.frame(out)
   }
-  out
+  on.exit(reader$close())  
```

Comment:
Move the `on.exit` right after the creation as failure can happen in between and you'll get a leak.